lib/config/defaults.js

```diff
@@ -709,14 +709,14 @@ const applyModuleDefaults = (
 		}
 		if (css) {
 			const cssRule = {
-				type: CSS_MODULE_TYPE,
+				type: CSS_MODULE_TYPE_AUTO,
 				resolve: {
 					fullySpecified: true,
 					preferRelative: true
 				}
 			};
 			const cssModulesRule = {
-				type: "css/module",
+				type: CSS_MODULE_TYPE_AUTO,
 				resolve: {
 					fullySpecified: true
 				}
```

Comment:
We should rewrite our logic:
```
const cssRule = {
  type: CSS_MODULE_TYPE,
  resolve: {
    fullySpecified: true,
    preferRelative: true
  }
};
const cssRuleAuto = {
  type: CSS_MODULE_TYPE_AUTO,
  resolve: {
    fullySpecified: true,
    preferRelative: true
  }
};
const cssModulesRule = {
  type: CSS_MODULE_TYPE_AUTO,
  resolve: {
    fullySpecified: true
  }
};
// Handle all CSS, and we will check is it modules inside parser
rules.push({
  test: /\.css$/i,
  ...cssRule,
});
rules.push({
  mimetype: "text/css+module",
  // Should be always only modules
  ...cssModulesRule
});
rules.push({
  mimetype: "text/css",
  // Should be always pure CSS
  ...cssRule
});
```

And let's add a test case for the each rule (for some we already have)